analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CloudNative/AzureFunctionsCatchExceptions.cs

```diff
@@ -0,0 +1,534 @@
+    {
+        try
+        {
+        }
+        catch (NullReferenceException)
+        {
+            DoSomething();  // This is compliant, because it can be the desired logging.
+        }
+    }
+
+    private static string DoSomething() => null;    // Compliant, btw
+}
+
+public static class AttributeVariants
+{
+    [FunctionNameAttribute("ClassName")]
+    public static void ClassName()          // Noncompliant
+    {
+        DoSomething();
+    }
+
+    [Microsoft.Azure.WebJobs.FunctionNameAttribute("NamespaceName")]
+    public static void NamespaceName()      // Noncompliant
+    {
+        DoSomething();
+    }
+
+    [global::Microsoft.Azure.WebJobs.FunctionNameAttribute("GlobalName")]
+    public static void GlobalName()         // Noncompliant
+    {
+        DoSomething();
+    }
+
+    private static void DoSomething() { }
+}
+
+public static class CatchScenarios
+{
+    [FunctionName("Sample")]
+    public static void OuterCatch()
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch (Exception ex)
+        {
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void OuterCatchFullName()
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch (System.Exception ex)
+        {
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void OuterCatchGlobalName()
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch (global::System.Exception ex)
+        {
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void OuterCatchNoVariable()
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch (Exception)
+        {
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void OuterCatchNoType()
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch
+        {
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void OuterCatchSpecific()   // Noncompliant
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch (NullReferenceException ex)
+        {
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void OuterCatchSpecificAndAll()
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch (NullReferenceException ex)
+        {
+        }
+        catch (Exception ex)
+        {
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void OuterCatchSpecificAndAll_When()   // Noncompliant
+    {
+        try
+        {
+            var i = 42;
+            DoSomething();
+        }
+        catch (NullReferenceException ex)
+        {
+        }
+        catch (Exception ex) when (ex is ArgumentNullException)
```

Comment:
This is a bit unfortunate because of the abuse of exception filters for logging purposes.

Reply:
I'd keep it like this because the rule is not in SonarWay. We can't tell what kind of filtering there will be. If any.

We can change the decision if we receive feedback about it.